presto-orc/src/main/java/io/prestosql/orc/metadata/OrcMetadataWriter.java

```diff
@@ -98,7 +108,7 @@ public int writeFooter(SliceOutput output, Footer footer)
             throws IOException
     {
         OrcProto.Footer footerProtobuf = OrcProto.Footer.newBuilder()
-                .setWriter(PRESTO_WRITER_ID)
+                .setWriter(useLegacyVersion ? HIVE_WRITER_ID : PRESTO_WRITER_ID)
```

Comment:
Maybe in legacy we shouldn't write writer id at all?

Reply:
Good idea. That field wasn't added until writer version 6.